src/mpi-base.jl

```diff
@@ -300,6 +300,55 @@ function Comm_split_type(comm::Comm,split_type::Integer,key::Integer;info::Info=
     MPI.Comm(newcomm[])
 end
 
+function Dims_create!(nnodes::Integer, ndims::Integer, dims::Vector{Cint})
```

Comment:
Arrays that get passed directly into MPI can be `MPIBufferType{T}`, which allows SubArrays and other things.

Reply:
OK, I changed it to `MPIBuffertype{T}`; also, in the other functions.